test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
can you also add a test for the case when FactoriedService depend on another service to make sure `params` work as expected